tests/test_utils.py

```diff
@@ -120,10 +119,42 @@ def __init__(self, size):
         self.linear = torch.nn.Linear(size, 1)
 
     def forward(self, prepared_batch):
+        input_sample = SampleList(prepared_batch)
         batch = prepared_batch[DATA_ITEM_KEY]
         output = self.linear(batch)
-        loss = torch.nn.MSELoss()(output, batch)
-        return {"losses": {"loss": loss}, "logits": output}
+        loss = torch.nn.MSELoss()(-1 * output, batch)
+        return {"losses": {"loss": loss}, "logits": output, "input_batch": input_sample}
+
+
+class SimpleLightningModel(pl.LightningModule):
+    def __init__(self, size, config=None):
+        super().__init__()
+        self.model = SimpleModel(size)
+        self.config = config
+
+    def forward(self, prepared_batch):
+        return self.model(prepared_batch)
+
+    def training_step(self, batch, batch_idx, *args, **kwargs):
+        output = self(batch)
+        output["loss"] = output["losses"]["loss"]
+        return output
+
+    def configure_optimizers(self):
+        if self.config is None:
+            return torch.optim.Adam(self.parameters(), lr=0.01)
+        else:
+            from mmf.utils.build import build_optimizer, build_scheduler
+
+            optimizer = build_optimizer(self, self.config)
+            if self.config.training.lr_scheduler:
+                lr_scheduler = build_scheduler(optimizer, self.config)
+                return {
+                    "optimizer": optimizer,
+                    "lr_scheduler": {"scheduler": lr_scheduler, "interval": "step"},
+                }
+            else:
+                return optimizer
```

Comment:
Might be better to refactor this to a function as it is used in BaseModel as well.